org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/CanonicalResourceManagerTests.java

```diff
@@ -927,4 +929,877 @@ public void testSupplements() {
+    CanonicalResourceManager.CachedCanonicalResource cc1 = csm.new CachedCanonicalResource<CodeSystem>(cp1, pi1);
+    csm.see(cc1);
+
+    CodeSystem cs = csm.get("http://test", null);
+    Assertions.assertEquals("2", cs.getId()); // Versioned wins
+  }
+
+  @Test
+  public void testBothUnversioned_MostRecentWins() throws ParseException {
+    CanonicalResourceManager<CodeSystem> csm = new CanonicalResourceManager<CodeSystem>(false, false);
+
+    CodeSystem cs1 = makeCodeSystem("1", "http://test", null);
+    CanonicalResourceProxy cp1 = new DeferredLoadTestResource(cs1);
+    PackageInformation pi1 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc1 = csm.new CachedCanonicalResource<CodeSystem>(cp1, pi1);
+    csm.see(cc1);
+
+    CodeSystem cs2 = makeCodeSystem("2", "http://test", null);
+    CanonicalResourceProxy cp2 = new DeferredLoadTestResource(cs2);
+    PackageInformation pi2 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc2 = csm.new CachedCanonicalResource<CodeSystem>(cp2, pi2);
+    csm.see(cc2);
+
+    CodeSystem cs = csm.get("http://test", null);
+    Assertions.assertEquals("2", cs.getId()); // Most recent wins
+  }
+
+  // =============================================
+  // PACKAGE COMPARISON TESTS
+  // =============================================
+
+  @Test
+  public void testSameVersion_PackagedVsUnpackaged_PackagedWins() throws ParseException {
+    CanonicalResourceManager<CodeSystem> csm = new CanonicalResourceManager<CodeSystem>(false, false);
+
+    CodeSystem cs1 = makeCodeSystem("1", "http://test", "2.0.1");
+    CanonicalResourceProxy cp1 = new DeferredLoadTestResource(cs1);
+    PackageInformation pi1 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc1 = csm.new CachedCanonicalResource<CodeSystem>(cp1, pi1);
+    csm.see(cc1);
+
+    CodeSystem cs2 = makeCodeSystem("2", "http://test", "2.0.1");
+    CanonicalResourceProxy cp2 = new DeferredLoadTestResource(cs2);
+    CanonicalResourceManager.CachedCanonicalResource cc2 = csm.new CachedCanonicalResource<CodeSystem>(cp2, null); // No package
+    csm.see(cc2);
+
+    CodeSystem cs = csm.get("http://test", "2.0.1");
+    Assertions.assertEquals("2", cs.getId()); // Packaged wins
+  }
+
+  @Test
+  public void testSameVersion_BothUnpackaged_MostRecentWins() throws ParseException {
+    CanonicalResourceManager<CodeSystem> csm = new CanonicalResourceManager<CodeSystem>(false, false);
+
+    CodeSystem cs1 = makeCodeSystem("1", "http://test", "2.0.1");
+    CanonicalResourceProxy cp1 = new DeferredLoadTestResource(cs1);
+    CanonicalResourceManager.CachedCanonicalResource cc1 = csm.new CachedCanonicalResource<CodeSystem>(cp1, null);
+    csm.see(cc1);
+
+    CodeSystem cs2 = makeCodeSystem("2", "http://test", "2.0.1");

```

Comment:
Nitpick: To make most-recent clearer, maybe swap the IDs to something different. If 2 is added before 1, and then 1 is the outcome, it's clearer that the id isn't a factor

Reply:
don't follow, but is it worth it?